docs/api.rst

```diff
@@ -1202,6 +1202,15 @@ to handle it, which defaults to print a traceback and ignoring the exception.
     :param execution: The object containing the execution information.
     :type execution: :class:`AutoModerationActionExecution`
 
+.. function:: on_application_command(interaction)
+
+    Called when an application command is executed
+
+    .. versionadded:: 2.4
+
+    :param interaction: The interaction object containing the command information.
```

Comment:
As I told you in a previous review comment, the interaction object at the time of dispatch doesn't actually contain the application command object.